uit/uit.py

```diff
@@ -726,6 +745,92 @@ def _parse_hpc_output(cls, output, as_df, columns=None, delimiter=None, delimite
             return cls._as_df(rows, columns)
         return rows
 
+    def _debug_uit(self, local_vars):
+        """ Show information about and around UIT+ calls for debug logging
+
+        It can be called from any UIT Client method right after using requests.post().
+        The recommended way to call this is:
+            debug_start_time = time.perf_counter()
+            r = requests.post(...)
+            logger.debug(self._debug_uit(locals()))
+
+        It will not run if DEBUG logging is not enabled since this code is not perfect,
+        and nobody wants to see debug log code cause exceptions in production.
+        It doesn't run logger.debug() itself so that the parent function will show in the logs.
+        """
+        if not logger.isEnabledFor(logging.DEBUG):
+            return
+
+        debug_end_time = time.perf_counter()
+        time_text = f"{debug_end_time - local_vars['debug_start_time']:.2f}s"
+
+        debug_header = f" {FG_RED}time={time_text}{ALL_OFF}    node={self.login_node}"
+
+        try:
+            resp = local_vars['r'].json()
+        except simplejson.errors.JSONDecodeError:
+            resp = {}
+
+        if resp.get('exitcode') is not None:
+            debug_header += f"    rc={resp.get('exitcode')}"
+        debug_header += f"    username={self.username}"
+
+        # stdout and stderr will only show up for call() and if they contain text
+        nice_stdout = ""
+        if resp.get('stdout'):
+            nice_stdout = "\n  stdout="
+            if len(resp.get('stdout')) > 500:
+                nice_stdout += "'" + resp.get('stdout')[:500].replace('\n', '\\n') + \
+                               f"'  <len:{len(resp.get('stdout'))}>"
+            else:
+                nice_stdout += "'" + resp.get('stdout').replace('\n', '\\n') + "'"
+
+        nice_stderr = ""
+        if resp.get('stderr'):
+            nice_stderr = "\n  stderr="
+            if len(resp.get('stderr')) > 500:
+                nice_stderr += "'" + resp.get('stderr')[:500].replace('\n', '\\n') + \
+                               f"'  <len:{len(resp.get('stderr'))}>"
+            else:
+                nice_stderr += "'" + resp.get('stderr').replace('\n', '\\n') + "'"
+
+        # Show only relevant function calls and ignore standard library
+        only_show_stacktrace = ['pyuit',
```

Comment:
@araglu PyUIT is designed to be a generic library and is used by other projects, so we shouldn't have any project specific references in there (e.g. helios, tethys, uit_plus, etc.). Could we make this list configurable? Maybe the default is just `['pyuit']`, but then it can check the `self.config_file` for a user specified list to use instead.

Reply:
I changed it to grab a list from the config_file.
Thanks for the suggestion! I wanted to do something that would only show dev code automagically such as filtering out site-packages, but I doubt that would work on a dev server, and I may want to peek inside one or two libraries occasionally.